### PR TITLE
added docker-feature

### PR DESCRIPTION
**Pull request by serhii-shyian**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -31,3 +31,6 @@ build/
 
 ### VS Code ###
 .vscode/
+
+### Docker ###
+.env
```
Comment (MentorMateAcademy) [inline .gitignore]: why do you ignore env.file?

**Review by MentorMateAcademy — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (serhii-shyian) [reply .gitignore]: I thought it was a good idea) I will fix it now.

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  mysqldb:
+    platform: linux/arm64
+    image: mysql:8.4.0
+    restart: unless-stopped
+    env_file: ./.env
+    environment:
+      - MYSQL_ROOT_PASSWORD=$MYSQL_ROOT_PASSWORD
+      - MYSQL_DATABASE=$MYSQL_DATABASE
+    ports:
+      - $MYSQL_DOCKER_PORT:$MYSQL_LOCAL_PORT
```
Comment (sarakhmen) [inline docker-compose.yml]: the first one value stated here corresponds to the local port that will be used to access the service outside the docker. The second one - port within the docker. So please consider it

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  app:
+    depends_on:
+      mysqldb:
+        condition: service_healthy
+    restart: unless-stopped
+    image: bookstore-app
+    build: .
+    env_file: ./.env
+    ports:
+      - $SPRING_DOCKER_PORT:$SPRING_LOCAL_PORT
```
Comment (sarakhmen) [inline docker-compose.yml]: same here

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      - $DEBUG_PORT:$DEBUG_PORT
+    environment:
+      SPRING_APPLICATION_JSON: '{
+        "spring.datasource.url": "$MYSQL_DOCKER_PATH:$MYSQL_LOCAL_PORT/$MYSQL_DATABASE",
```
Comment (sarakhmen) [inline docker-compose.yml]: and here you should specify docker port as your both services are run inside the docker

**Review by sarakhmen — CHANGES_REQUESTED** [review]
let's improve your solution :)

File: docker-compose.yml (hunk 2 above)
Comment (serhii-shyian) [reply docker-compose.yml]: Ok, I fix it, please check my solution

**Review by Elena-Bruyako — APPROVED** [review]
looks great

**Pull request merged**